File: python/bubble_sort.py
import sys
import time
import random

def bubble_sort(arr):
    n = len(arr)
    for i in range(n - 1):
        for j in range(n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

def generate_array(n):
    random.seed(42)  # Fixed seed for reproducible results
    return [random.randint(0, 9999) for _ in range(n)]

if len(sys.argv) < 2:
    print('Usage: bubble_sort <n>')
    sys.exit(1)

try:
    n = int(sys.argv[1])
except ValueError:
    print('Invalid number:', sys.argv[1])
    sys.exit(1)

arr = generate_array(n)

start = time.time()
bubble_sort(arr)
end = time.time()

time_ms = int((end - start) * 1000)

print(f'Python: bubble_sort({n}) = sorted')
print(f'Time: {time_ms}ms')

File: swift/bubble_sort.swift
import Foundation

func bubbleSort(_ arr: inout [Int]) {
    let n = arr.count
    for i in 0..<(n - 1) {
        for j in 0..<(n - i - 1) {
            if arr[j] > arr[j + 1] {
                arr.swapAt(j, j + 1)
            }
        }
    }
}

func generateArray(_ n: Int) -> [Int] {
    var rng = SystemRandomNumberGenerator()
    rng = SystemRandomNumberGenerator() // Reset with fixed behavior
    return Array(0..<n).map { _ in Int.random(in: 0..<10000, using: &rng) }
}

// Simple seeded random for reproducible results
func generateArraySeeded(_ n: Int) -> [Int] {
    srand48(42) // Fixed seed
    return Array(0..<n).map { _ in Int(drand48() * 10000) }
}

let args = CommandLine.arguments
if args.count < 2 {
    print("Usage: bubble_sort <n>")
    exit(1)
}

guard let n = Int(args[1]) else {
    print("Invalid number: \(args[1])")
    exit(1)
}

var arr = generateArraySeeded(n)

let start = CFAbsoluteTimeGetCurrent()
bubbleSort(&arr)
let end = CFAbsoluteTimeGetCurrent()

let timeMs = Int((end - start) * 1000)

print("Swift: bubble_sort(\(n)) = sorted")
print("Time: \(timeMs)ms")